google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ResourceLoadingModelItem.java

```diff
@@ -16,12 +16,46 @@
 
 package com.google.cloud.tools.intellij.resources;
 
+import com.sun.istack.NotNull;
+
 import javax.swing.tree.DefaultMutableTreeNode;
 
 /**
- * This model item is shown in the project selector when a resource manager call is outstanding.
+ * This model item is shown when a resource manager call is outstanding.
  */
 class ResourceLoadingModelItem extends DefaultMutableTreeNode {
 
 }
 
+/**
+ * This model item is shown when a resource manager call returns an empty result.
+ */
+class ResourceEmptyModelItem extends DefaultMutableTreeNode {
```

Comment:
The bigger point is that if you look at RepositorySelectorRenderer#getTreeCellRendererComponent its checking the type of DefaultMutableTreeNode so it can then decide which UX to show.